tapchannel/aux_sweeper.go

```diff
@@ -324,16 +318,6 @@ func (a *AuxSweeper) createSweepVpackets(sweepInputs []*cmsg.AssetOutput,
 				"assets: %w", err)
 		}
 
-		// Next before we sign, we'll make sure to update the witness
-		// of the prev asset's root asset. Otherwise, we'll be signing
-		// the wrong input leaf.
-		vIn := vPackets[idx].Inputs[0]
-		if vIn.Asset().HasSplitCommitmentWitness() {
-			//nolint:lll
-			rootAsset := vIn.Asset().PrevWitnesses[0].SplitCommitment.RootAsset
```

Comment:
Hmm, I recall this was one of those bugs that took be a while to track down properly. 

The root asset (funding output) in this case needs to have the updated witness w/ the control block in tact, as we don't apply the segwit encoding to root assets stored as part of a split commitment.

Reply:
Hmm, yeah, I thought this was probably the case. But I removed these lines and ran a full integration test suite and nothing broke. Is it possible that this was fixed indirectly in a later version by adding the witness earlier in the flow? Because I checked and the commitment transaction should already have the witness (both in the root asset as well as in each split commitment's root asset copy). But I will double check again.

Reply:
Okay, I found exactly why this isn't needed anymore.
This was fixed with c072d3f2, which was in #1154.
Which was quite a while after de5bed63 (in #938) which introduced the temporary fix in the first place.
So it seems like it was indeed needed at some point but could've been removed in #1154 already.